pyvista/core/dataobject.py

```diff
@@ -873,16 +887,16 @@ def _unserialize_pyvista_pickle_format(self, state):
             reader.Update()
 
         elif pickle_format.lower() == 'legacy':
-            reader = _vtk.vtkDataSetReader()  # type: ignore[assignment]
+            reader = _vtk.vtkDataSetReader()
             reader.ReadFromInputStringOn()
             if isinstance(vtk_serialized, bytes):
-                reader.SetBinaryInputString(vtk_serialized, len(vtk_serialized))  # type: ignore[attr-defined]
+                reader.SetBinaryInputString(str(vtk_serialized), len(vtk_serialized))
```

Comment:
this line may need to be reverted. Type was `bytes` but cast as `str` since this method is typed for `str` inputs. Some pickle tests are failing though, this could be the culprit